### DISCUSSION TITLE
Eslint false positive on useEffectEvent?

### DISCUSSION THREAD

**Discussion by aeharding** [reactwg/react-compiler#27] [Something went wrong]
Hi! I have the eslint plugin showing an error for the following code snippet (trimmed for brevity):

https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEwAOgHYEFwwICGeCAwhKfQB54A0ZFUYjNAGwEAjGnADWXcgV6NmbTtxl8AogDM1COIumyAsggwQpPPgCUEak8oQBlPHQTWErLAhiYELQQH1Z6zW0VADcvQhowGwCtPBCwqQBfAjUYCAwCYhBqMTxMgG4yMhccfGSoUm00ZhsANTQAEwQIAAVcBwEAOQhGgAoweGQCMDwPUgBzAEpB1vxBLsaCAB8CYIgGoiU4ZmGh6HgEAFUGizUCAF4bE57ytABHKAQASXqeiYmC6S3SHeACMYQ8DN2vMEAAxXC2eAcSgCWikKBYIFzbpgiHwR5qDoIBCNerQ4INJonSJJM5KUxyFguPA9OqNFptQRMKnsd6FT7bQiNYapACeOPOl0sPTUgj4bNIm05fwBSM6KLiLEF-g0MUVNNe5wAfBtpBQ0KcetyRhB+fUAHRwWDUFgTAjUPCwUgfckywGM+WNcEwSFwPpQ3awRBHeonS3WsISigJCVS76EOCwmjwxEekEYrE4nHq5WqVVBUIsHqas46kh6gjGvk48MwG2EC4jB4fCgURNwhFykHe30Z7G4-1waGQIOHY6WWv1qMEGMu3R5wIau2l3WtysIHmmmtWuthQWigR8Fut-7u2aehDq15ztcOp0wjup8-pzH97OFvDHhLQgDaAF0JQSMgQASIA

And I'm getting the error:

```InvalidReact: Ref values (the `current` property) may not be accessed during render. (https://react.dev/reference/react/useRef) (19:23)```

My question, is this a valid complaint (am I doing something wrong)? Or is it a problem/limitation with the eslint plugin?

**Comment by poteto** [comment]
Thanks for flagging this! This looks like a bug in how the compiler understands import statements. This isn't specifically about `useEffectEvent` but because it's prefixed with `experimental_` in the original identifier. The compiler is mistakenly reading it as a non-hook because of that prefix.

**Reply by poteto** [reply]
Fixed in https://github.com/facebook/react/pull/31384
